bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/AccountHandler.java

```diff
@@ -591,28 +598,27 @@ private void refreshData() {
 
     public @Nullable Device findDeviceJson(@Nullable String serialNumber) {
         Device result = null;
-        if (StringUtils.isNotEmpty(serialNumber)) {
+        if (serialNumber != null && !serialNumber.isEmpty()) {
             Map<String, Device> jsonSerialNumberDeviceMapping = this.jsonSerialNumberDeviceMapping;
             result = jsonSerialNumberDeviceMapping.get(serialNumber);
         }
         return result;
     }
 
     public @Nullable Device findDeviceJsonBySerialOrName(@Nullable String serialOrName) {
-        if (StringUtils.isNotEmpty(serialOrName)) {
-            Map<String, Device> currentJsonSerialNumberDeviceMapping = this.jsonSerialNumberDeviceMapping;
-            for (Device device : currentJsonSerialNumberDeviceMapping.values()) {
-                if (StringUtils.equalsIgnoreCase(device.serialNumber, serialOrName)) {
-                    return device;
-                }
-            }
-            for (Device device : currentJsonSerialNumberDeviceMapping.values()) {
-                if (StringUtils.equalsIgnoreCase(device.accountName, serialOrName)) {
-                    return device;
-                }
-            }
+        if (serialOrName == null || serialOrName.isEmpty()) {
+            return null;
+        }
+
+        Optional<Device> device = this.jsonSerialNumberDeviceMapping.values().stream().filter(
+                d -> serialOrName.equalsIgnoreCase(d.serialNumber) || serialOrName.equalsIgnoreCase(d.accountName))
+                .findFirst();
+
+        if (device.isPresent()) {
+            return device.get();
+        } else {
+            return null;
         }
```

Comment:
```suggestion
       return this.jsonSerialNumberDeviceMapping.values().stream().filter(
                d -> serialOrName.equalsIgnoreCase(d.serialNumber) || serialOrName.equalsIgnoreCase(d.accountName))
                .findFirst().orElse(null);
```

Reply:
Turns into: Null type mismatch: required 'JsonDevices.@NonNull Device' but the provided value is null
I do not have any experience about this @NonNull and @Nullable Annotations, sadly.

Reply:
Then leave it as it is. Obviously it's the same but the null annotations do not properly handle `Optional`s.